scripts/split_identify_fasta.py

```diff
@@ -11,35 +11,44 @@
 parser.add_argument("--Ns", type = int, default = 5000, help = "Number of Ns to split by [5000]")
 
 args = parser.parse_args()
+def read_fasta(fp):
+    name, seq = None, []
+    for line in fp:
+        line = line.rstrip()
+        if line.startswith(">"):
+            if name: yield (name, ''.join(seq))
+            name, seq = line, []
+        else:
+            seq.append(line)
+    if name: yield (name, ''.join(seq))
+     
 
 with open(args.outputfile, "w+") as outf:
     with open(args.inputfile) as f:
-        for linenr, line in enumerate(f):
-            if linenr % 2 == 0:
-                rid = line.rstrip().lstrip(">")
-            else:
-                if line[0] not in {'A', 'G', 'C', 'T', 'N'}:
-                    print("Error found in input file. Make sure the input is in correct fasta format.")
-                    sys.exit()
-                print(rid + "... ", end = "")
-
-                s = line.rstrip().upper().lstrip("N").rstrip("N")
-                matches = re.finditer("N{" + str(args.Ns) + ",}", s)
-                start = 0
-                partnr = 1
-                for m in matches:
-                    nstart, nstop = m.span()
-                    #print(s[m.span()[1]])
-                    part = s[start:nstart]
-
-                    outf.write(">" + args.prefix + "_" + rid + "_" + str(partnr) + "\n")
-                    outf.write(part + "\n")
-
-                    partnr += 1
-                    start = nstop
-                part = s[start:]
-                outf.write(">" + args.prefix + "_" + rid + "_" + str(partnr) + "\n")
+        for fid, seqt in read_fasta(f):
+            seq = seqt.upper()
+            if seq[0] not in {'A', 'G', 'C', 'T', 'N'}:
+                print("Error found in input file. Make sure the input is in correct fasta format.")
+                sys.exit()
+            print(fid + "... ", end = "")
+
+            s = seq.rstrip().upper().lstrip("N").rstrip("N")
+            matches = re.finditer("N{" + str(args.Ns) + ",}", s)
+            start = 0
```

Comment:
I always worry when I see this in python scripts, as it reminds me of C code----there are normally quicker ways algorithmically to do things besides iterating like this. 

Nonetheless, it works :) You might want to investigate whether this part could be more efficient---I don't have strong ideas at the moment.

Reply:
It's not something that will be used very often. I am certain there are faster ways to do this, but the bottleneck of the pipeline will definitely be elsewhere.

Reply:
fair enough